cmd/autoheal/alerts_worker.go

```diff
@@ -120,94 +128,74 @@ func (h *Healer) cancelHealing(alert *alertmanager.Alert) error {
 	return nil
 }
 
-func (h *Healer) checkConditions(rule *monitoring.HealingRule, alert *alertmanager.Alert) bool {
+func (h *Healer) checkRule(rule *monitoring.HealingRule, alert *alertmanager.Alert) (matches bool, err error) {
 	glog.Infof(
-		"Checking conditions of rule '%s' for alert '%s'",
+		"Checking rule '%s' for alert '%s'",
 		rule.ObjectMeta.Name,
 		alert.Name(),
 	)
-	if rule.Spec.Conditions != nil && len(rule.Spec.Conditions) > 0 {
-		for i := 0; i < len(rule.Spec.Conditions); i++ {
-			if !h.checkCondition(&rule.Spec.Conditions[i], alert) {
-				return false
-			}
-		}
+	matches, err = h.checkMap(alert.Labels, rule.Labels)
+	if !matches || err != nil {
+		return
 	}
-	return true
-}
-
-func (h *Healer) checkCondition(condition *monitoring.HealingCondition, alert *alertmanager.Alert) bool {
-	matched, err := regexp.MatchString(condition.Alert, alert.Name())
-	if err != nil {
-		glog.Errorf(
-			"Error while checking if alert name '%s' matches pattern '%s': %s",
-			alert.Name(),
-			condition.Alert,
-			err.Error(),
-		)
-		matched = false
+	matches, err = h.checkMap(alert.Annotations, rule.Annotations)
+	if !matches || err != nil {
+		return
 	}
-	return matched
+	return
 }
 
-func (h *Healer) runActions(rule *monitoring.HealingRule, alert *alertmanager.Alert) error {
-	if rule.Spec.Actions != nil && len(rule.Spec.Actions) > 0 {
-		glog.Infof(
-			"Running actions of healing rule '%s' for alert '%s'",
-			rule.ObjectMeta.Name,
-			alert.Name(),
-		)
-		for i := 0; i < len(rule.Spec.Actions); i++ {
-			err := h.runAction(rule, &rule.Spec.Actions[i], alert)
-			if err != nil {
-				return err
+func (h *Healer) checkMap(values, patterns map[string]string) (matches bool, err error) {
+	if patterns != nil {
+		for key, pattern := range patterns {
+			if values != nil {
+				value, ok := values[key]
+				if ok {
+					matches, err = regexp.MatchString(pattern, value)
+					if !matches || err != nil {
+						return
+					}
+				}
 			}
 		}
-	} else {
-		glog.Warningf(
-			"Healing rule '%s' has no actions, will have no effect on alert '%s'",
-			rule.ObjectMeta.Name,
-			alert.Name(),
-		)
 	}
-
-	return nil
+	matches = true
```

Comment:
shouldn't matches be false in this case?

Reply:
I think it should be `true`. The idea is that the method exists (returning `false`) as soon as it detects that the alert doesn't match. If it reaches this line then it means that the alert matched. In particular if the rule is empty then it will match all alerts, regardless of the labels of annotations that they have.

However, this method doesn't work correctly in some cases. In the next version of the pull request I am improving it and adding some tests to make sure that it works as expected.